network/dag/parser.go

```diff
@@ -108,15 +111,18 @@ func parseContentType(document *document, headers jws.Headers, _ *jws.Message) e
 	return nil
 }
 
-// parseSigningKey parses, validates and sets the document signing key (`jwk`) or key ID (`kid`).
-func parseSigningKey(document *document, headers jws.Headers, _ *jws.Message) error {
+// parseSignatureParams parses, validates and sets the document signing key (`jwk`) or key ID (`kid`).
+func parseSignatureParams(document *document, headers jws.Headers, _ *jws.Message) error {
```

Comment:
it feels as if the `document` should wrap the JWS and provide functions instead of fields.